src/unstated.js

```diff
@@ -64,24 +64,45 @@ export type SubscribeProps<Containers: ContainersType> = {
   ) => Node
 };
 
-type SubscribeState = {};
+type SubscribeUpdaterProps = {
+  instances: Array<ContainerType>,
+  children: (
+    ...instances: $TupleMap<Containers, <C>(Class<C> | C) => C>
+  ) => Node
+};
+type SubscribeUpdaterState = {};
 
 const DUMMY_STATE = {};
 
-export class Subscribe<Containers: ContainersType> extends React.Component<
-  SubscribeProps<Containers>,
-  SubscribeState
+class SubscribeUpdater<Containers: ContainersType> extends React.Component<
```

Comment:
I'm not a big fan of splitting the `Subscribe` component in two components, but because of the context API, it looked like the cleaner method to handle the lifecycle of the instances (subscribe to new instances, unsubscribe from previous one when it render, etc).